staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -137,159 +208,90 @@ func New(opts OIDCOptions) (*OIDCAuthenticator, error) {
 		TLSClientConfig: &tls.Config{RootCAs: roots},
 	})
 
-	authenticator := &OIDCAuthenticator{
-		issuerURL:       opts.IssuerURL,
-		trustedClientID: opts.ClientID,
-		usernameClaim:   opts.UsernameClaim,
-		usernamePrefix:  opts.UsernamePrefix,
-		groupsClaim:     opts.GroupsClaim,
-		groupsPrefix:    opts.GroupsPrefix,
-		httpClient:      &http.Client{Transport: tr},
-	}
-
-	// Attempt to initialize the authenticator asynchronously.
-	//
-	// Ignore errors instead of returning it since the OpenID Connect provider might not be
-	// available yet, for instance if it's running on the cluster and needs the API server
-	// to come up first. Errors will be logged within the client() method.
-	go func() {
-		defer runtime.HandleCrash()
-		authenticator.client()
-	}()
-
-	return authenticator, nil
-}
-
-// Close stops all goroutines used by the authenticator.
-func (a *OIDCAuthenticator) Close() {
-	a.mu.Lock()
-	defer a.mu.Unlock()
-
-	if a.close != nil {
-		a.close()
-	}
-	return
-}
-
-func (a *OIDCAuthenticator) client() (*oidc.Client, error) {
-	// Fast check to see if client has already been initialized.
-	if client := a.oidcClient.Load(); client != nil {
-		return client.(*oidc.Client), nil
-	}
+	client := &http.Client{Transport: tr, Timeout: 15 * time.Second}
```

Comment:
this is the client used in the background to fetch discovery docs and public keys? what was the timeout before?